lib/galaxy/tools/parameters/basic.py

```diff
@@ -936,6 +936,11 @@ def from_json(self, value, trans, other_values={}, require_legal_value=True):
             if is_runtime_value(value):
                 return None
             if value not in legal_values and require_legal_value:
+                if isinstance(self, ColumnListParameter):
```

Comment:
Rather than isinstance you should move this code to here: https://github.com/galaxyproject/galaxy/blob/f7bba746b1d33693c2f4f3f59d829b54756da2ba/lib/galaxy/tools/parameters/basic.py#L1339

Reply:
I don't think that's better (https://github.com/galaxyproject/galaxy/issues/10918#issuecomment-749638291)